pkg/logs/processor/processor.go

```diff
@@ -138,7 +225,26 @@ func (p *Processor) processMessage(msg *message.Message) {
 		p.outputChan <- msg
 		p.pipelineMonitor.ReportComponentIngress(msg, metrics.StrategyTlmName, p.instanceID)
 	}
+}
+
+// filterMRFMessages applies an MRF tag to messages that should be sent to MRF
+// destinations
+func (p *Processor) filterMRFMessages(msg *message.Message) {
+	serviceAllowlist := p.failoverConfig.failoverServiceAllowlist
+	minLevel := p.failoverConfig.failoverMinLevel
 
+	// Tag the message for failover if:
+	// 1. No allowlists are configured (failover all).
+	// 2. The message service is in the service allowlist.
+	// 3. The message status is at or above the configured minimum log level.
+	noFilters := len(serviceAllowlist) == 0 && minLevel < 0
+	_, serviceMatch := serviceAllowlist[msg.Origin.Service()]
+	msgSeverity, hasSeverity := severities[msg.GetStatus()]
+	statusMatch := minLevel >= 0 && hasSeverity && msgSeverity >= minLevel
+
+	if noFilters || serviceMatch || statusMatch {
+		msg.IsMRFAllow = true
+	}
```

Comment:
Because this is a hot path in the Agent, I would consider early returns here. You might win only a few ops, but they will matter, especially since the first if is the less costly and will be the most used branch.
E.g.
```suggestion
	if len(serviceAllowlist) == 0 && minLevel < 0 {
		msg.IsMRFAllow = true
		return
	}
	
	_, serviceMatch := serviceAllowlist[msg.Origin.Service()]
	if serviceMatch {
		msg.IsMRFAllow = true
		return
	}

	msgSeverity, hasSeverity := severities[msg.GetStatus()]
	statusMatch := minLevel >= 0 && hasSeverity && msgSeverity >= minLevel
	if statusMatch {
		msg.IsMRFAllow = true
	}
```